### ISSUE TITLE
8.2.19 (beta): Massive bugs, Bagnon refuses to work.

### ISSUE THREAD

**Issue by Voxxel** [tullamods/Bagnon#1045]
8.2.19 (beta) produces a lot of visible functionality bugs that forces me to turn off the addon everytime. After startup I get this one when I simply tried to grab one of the items and move it to another empty slot:

```
Message: Interface\AddOns\BagBrother\Startup.lua:78: attempt to call global 'HasKey' (a nil value)
Time: Mon Dec 16 22:29:15 2019
Count: 1
Stack: Interface\AddOns\BagBrother\Startup.lua:78: attempt to call global 'HasKey' (a nil value)
Interface\AddOns\BagBrother\Startup.lua:78: in function `UpdateData'
Interface\AddOns\BagBrother\Startup.lua:30: in function `?'
Interface\AddOns\BagBrother\Startup.lua:20: in function <Interface\AddOns\BagBrother\Startup.lua:20>

Locals: self = BagBrother {
 BANKFRAME_OPENED = <function> defined @Interface\AddOns\BagBrother\Events.lua:42
 ParseItem = <function> defined @Interface\AddOns\BagBrother\API.lua:47
 SetupEvents = <function> defined @Interface\AddOns\BagBrother\Startup.lua:49
 VOID_STORAGE_CLOSE = <function> defined @Interface\AddOns\BagBrother\Events.lua:68
 SaveBag = <function> defined @Interface\AddOns\BagBrother\API.lua:19
 PLAYER_EQUIPMENT_CHANGED = <function> defined @Interface\AddOns\BagBrother\Events.lua:31
 SaveEquip = <function> defined @Interface\AddOns\BagBrother\API.lua:40
 GUILDBANKFRAME_OPENED = <function> defined @Interface\AddOns\BagBrother\Events.lua:83
 Realm = <table> {
 }
 StartupCache = <function> defined @Interface\AddOns\BagBrother\Startup.lua:33
 PLAYER_MONEY = <function> defined @Interface\AddOns\BagBrother\Events.lua:35
 BANKFRAME_CLOSED = <function> defined @Interface\AddOns\BagBrother\Events.lua:46
 GUILDBANKFRAME_CLOSED = <function> defined @Interface\AddOns\BagBrother\Events.lua:87
 VOID_STORAGE_OPEN = <function> defined @Interface\AddOns\BagBrother\Events.lua:64
 Player = <table> {
 }
 RemoveEvent = <function> defined @Interface\AddOns\BagBrother\Startup.lua:89
 GUILDBANKBAGSLOTS_CHANGED = <function> defined @Interface\AddOns\BagBrother\Events.lua:95
 0 = <userdata>
 GUILD_ROSTER_UPDATE = <function> defined @Interface\AddOns\BagBrother\Events.lua:91
 UpdateData = <function> defined @Interface\AddOns\BagBrother\Startup.lua:69
 BAG_UPDATE = <function> defined @Interface\AddOns\BagBrother\Events.lua:25
}
(*temporary) = nil
(*temporary) = "attempt to call global 'HasKey' (a nil value)"
```

After that, I reloaded the game, and instantly get 2 more bugs, this one: 

```
Message: ...ce\AddOns\Bagnon\common\Wildpants\classes\tipped.lua:14: attempt to call method 'IsOwned' (a nil value)
Time: Mon Dec 16 22:34:08 2019
Count: 1
Stack: ...ce\AddOns\Bagnon\common\Wildpants\classes\tipped.lua:14: attempt to call method 'IsOwned' (a nil value)
...ce\AddOns\Bagnon\common\Wildpants\classes\tipped.lua:14: in function `OnLeave'
...ddOns\Bagnon\common\Wildpants\classes\moneyFrame.lua:31: in function <...ddOns\Bagnon\common\Wildpants\classes\moneyFrame.lua:31>

Locals: self = BagnonMoneyFrame1 {
 small = 1
 staticMoney = 16510250970
 trialErrorButton = BagnonMoneyFrame1TrialErrorButton {
 }
 moneyType = "PLAYER"
 info = <table> {
 }
 0 = <userdata>
 frame = <unnamed> {
 }
 overlay = <unnamed> {
 }
 vadjust = 0
}
(*temporary) = nil
(*temporary) = BattlePetTooltip {
 0 = <userdata>
 BorderBottomLeft = <unnamed> {
 }
 Owned = <unnamed> {
 }
 SpeedTexture = <unnamed> {
 }
 Background = <unnamed> {
 }
 HealthTexture = <unnamed> {
 }
 Speed = <unnamed> {
 }
 BorderTop = <unnamed> {
 }
 PetType = <unnamed> {
 }
 BorderTopLeft = <unnamed> {
 }
 BorderTopRight = <unnamed> {
 }
 BorderLeft = <unnamed> {
 }
 Power = <unnamed> {
 }
 PowerTexture = <unnamed> {
 }
 PetTypeTexture = <unnamed> {
 }
 BattlePet = <unnamed> {
 }
 Name = <unnamed> {
 }
 Health = <unnamed> {
 }
 Level = <unnamed> {
 }
 BorderBottom = <unnamed> {
 }
 BorderBottomRight = <unnamed> {
 }
 BorderRight = <unnamed> {
 }
}
(*temporary) = BagnonMoneyFrame1 {
 small = 1
 staticMoney = 16510250970
 trialErrorButton = BagnonMoneyFrame1TrialErrorButton {
 }
 moneyType = "PLAYER"
 info = <table> {
 }
 0 = <userdata>
 frame = <unnamed> {
 }
 overlay = <unnamed> {
 }
 vadjust = 0
}
(*temporary) = "attempt to call method 'IsOwned' (a nil value)"
```

and this one: 

```
Message: ...ce\AddOns\Bagnon\common\Wildpants\classes\tipped.lua:14: attempt to call method 'IsOwned' (a nil value)
Time: Mon Dec 16 22:37:03 2019
Count: 18
Stack: ...ce\AddOns\Bagnon\common\Wildpants\classes\tipped.lua:14: attempt to call method 'IsOwned' (a nil value)
...ce\AddOns\Bagnon\common\Wildpants\classes\tipped.lua:14: in function <...ce\AddOns\Bagnon\common\Wildpants\classes\tipped.lua:9>
(tail call): ?
...face\AddOns\Bagnon\common\Wildpants\classes\item.lua:215: in function <...face\AddOns\Bagnon\common\Wildpants\classes\item.lua:214>

Locals: self = ContainerFrame1Item13 {
 timeSinceUpgradeCheck = 0
 PostOnLoad = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:169
 BattlepayItemTexture = <unnamed> {
 }
 IconOverlay = <unnamed> {
 }
 BagStaticTop = <unnamed> {
 }
 Flash = <unnamed> {
 }
 GetItemContextMatchResult = <function> defined @Interface\FrameXML\ContainerFrame.lua:1463
 IconBorder = <unnamed> {
 }
 newitemglowAnim = <unnamed> {
 }
 icon = ContainerFrame1Item13IconTexture {
 }
 LevelLinkLockTexture = <unnamed> {
 }
 RegisterCallback = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:175
 UnregisterCallback = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:182
 PostOnHide = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:198
 NewItemTexture = <unnamed> {
 }
 Cooldown = ContainerFrame1Item13Cooldown {
 }
 BagStaticBottom = <unnamed> {
 }
 ItemContextOverlay = <unnamed> {
 }
 PostOnShow = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:189
 flash = <unnamed> {
 }
 itemContextChangedCallbackIsSet = true
 itemContextMatchResult = 3
 count = 6
 ExtendedSlot = <unnamed> {
 }
 ExtendedOverlay2 = <unnamed> {
 }
 hasItem = true
 flashAnim = <unnamed> {
 }
 info = <table> {
 }
 JunkIcon = <unnamed> {
 }
 SetMatchesSearch = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:202
 UpdateItemContextOverlay = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:222
 bag = 0
 frame = <unnamed> {
 }
 QuestBorder = ContainerFrame1Item13IconQuestTexture {
 }
 IconGlow = <unnamed> {
 }
 itemContextChangedCallback = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:170
 0 = <userdata>
 readable = false
 Count = ContainerFrame1Item13Count {
 }
 ExtendedOverlay = <unnamed> {
 }
 UpgradeIcon = <unnamed> {
 }
 GetMatchesSearch = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:207
 UpdateItemContextMatching = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:211
 searchOverlay = ContainerFrame1Item13SearchOverlay {
 }
}
(*temporary) = nil
(*temporary) = BattlePetTooltip {
 0 = <userdata>
 BorderBottomLeft = <unnamed> {
 }
 Owned = <unnamed> {
 }
 SpeedTexture = <unnamed> {
 }
 Background = <unnamed> {
 }
 HealthTexture = <unnamed> {
 }
 speciesID = 630
 Speed = <unnamed> {
 }
 BorderTop = <unnamed> {
 }
 PetType = <unnamed> {
 }
 BorderTopLeft = <unnamed> {
 }
 BorderTopRight = <unnamed> {
 }
 BorderLeft = <unnamed> {
 }
 Power = <unnamed> {
 }
 PowerTexture = <unnamed> {
 }
 PetTypeTexture = <unnamed> {
 }
 BattlePet = <unnamed> {
 }
 Name = <unnamed> {
 }
 Health = <unnamed> {
 }
 Level = <unnamed> {
 }
 BorderBottom = <unnamed> {
 }
 BorderBottomRight = <unnamed> {
 }
 BorderRight = <unnamed> {
 }
}
(*temporary) = ContainerFrame1Item13 {
 timeSinceUpgradeCheck = 0
 PostOnLoad = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:169
 BattlepayItemTexture = <unnamed> {
 }
 IconOverlay = <unnamed> {
 }
 BagStaticTop = <unnamed> {
 }
 Flash = <unnamed> {
 }
 GetItemContextMatchResult = <function> defined @Interface\FrameXML\ContainerFrame.lua:1463
 IconBorder = <unnamed> {
 }
 newitemglowAnim = <unnamed> {
 }
 icon = ContainerFrame1Item13IconTexture {
 }
 LevelLinkLockTexture = <unnamed> {
 }
 RegisterCallback = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:175
 UnregisterCallback = <function> defined @Interface\FrameXML\ItemButtonTemplate.lua:182
 PostOnHide = <function> defined @Int
```

But the most serious bug that makes me turn off Bagnon is when I close Auctioneer the half of my inventory is suddenly vanish
![WoWScrnShot_121619_222027](https://user-images.githubusercontent.com/3882152/70946161-fd780900-2056-11ea-8c84-5e87b8654b2c.jpg)
ed.
```

**Comment by Alwe969** [comment]
I have the same issues. Most annoying is the empy spaces, where you can't interact with your bags.

**Comment by Voxxel** [comment]
okay, it's not related to any other addons, I turned off everything but Bagnon and it's the same. The game loads, I open the bags, Bagnon shows up => I grab a random item put it to another random empty slot and that's it, all my items are frozen in place cant grab anything furthermore. Moreover, when I open the  ingame mail, the mail panel and the bagnon panel shows up but the bagnon panel is empty up to the item I relocated recently.

No any other addons enbled.
